src/Raven.Server/Monitoring/Snmp/Objects/Database/5.1/DatabaseLoadedCount.cs

```diff
@@ -0,0 +1,27 @@
+using System.Linq;
+using Lextm.SharpSnmpLib;
+using Raven.Server.Documents;
+
+namespace Raven.Server.Monitoring.Snmp.Objects.Documents
+{
+    public class DatabaseLoadedCount : ScalarObjectBase<Integer32>
+    {
+        private readonly DatabasesLandlord _landlord;
+
+        public DatabaseLoadedCount(DatabasesLandlord landlord)
+            : base("5.1.2")
+        {
+            _landlord = landlord;
+        }
+
+        protected override Integer32 GetData()
+        {
+            return new Integer32(GetCount(_landlord));
+        }
+
+        private static int GetCount(DatabasesLandlord landlord)
+        {
+            return landlord.DatabasesCache.Values.Count();
```

Comment:
I'm pretty sure that this takes a lock

Reply:
Actually, this is worse. It:
* Takes a lock
* Copy the entire values collection to an allocated buffer
* Return the buffer and then iterate on it only to discard it.

Reply:
I do not expect this to be called more often than once every 3-5 minutes. This is still a problem?

Reply:
Probably not, but better to use just the `Count` anyway